### PR TITLE
Add Parsec Threat Model

### PR DESCRIPTION
**Pull request by hug-dev**
This commit adds a page in the book describing the Parsec Threat Model
with a dataflow diagram.

cf parallaxsecond/parsec#89

[Rendered view](https://github.com/parallaxsecond/parsec-book/blob/91620f44c24e4f86a7152964643bdbad2d594089/src/threat_model/threat_model.md) of the threat model.

Signed-off-by: Hugues de Valon <hugues.devalon@arm.com>

### CODE REVIEW

File: src/threat_model/threat_model.md [hunk 1]
```diff
@@ -0,0 +1,305 @@
...
+### Application Identity - AS1
+
+The Client Library gets its Application Identity from the Identity Provider. It is signed by the
+IP's private key and sent by the Client Library on each request. Parsec verifies the Authentication
+Token using the IP's public key.
+
+**Confidentiality** : if known (the authentication token), an attacker could impersonate a specific
+application and execute operations in their name.
+
+*Not applicable for deployments with no Identity Provider.*
```
Comment (ionut-arm) [inline src/threat_model/threat_model.md]: Maybe something like 
> In a deployment without an Identity Provider tenants will still have an identity in the form of an application name, but all entities in the system are trusted not to use identities of other tenants and so confidentiality is not critical.

File: src/threat_model/threat_model.md [hunk 2]
```diff
@@ -0,0 +1,305 @@
...
+administrator.
+
+## Attackers
+
+Each dataflow is analysed from an attacker's perspective using STRIDE method. Nothing is supposed on
+the different components.
+
+In the following tables are present the type of each possible threat, its description, its
+mitigation status and the assets impacted. A threat can be unmitigated (U), mitigated (M) or
+operational (O). The assumptions context applies for all threats but when one of them is
```
Comment (ionut-arm) [inline src/threat_model/threat_model.md]: I'd rephrase it to 
> mitigated within the service (M) or mitigated through operational requirements (O)

File: src/threat_model/threat_model.md (hunk 1 above)
Comment (hug-dev) [reply src/threat_model/threat_model.md]: Yes good comment! Will add to clarify.

File: src/threat_model/threat_model.md (hunk 2 above)
Comment (hug-dev) [reply src/threat_model/threat_model.md]: Makes sense, will change.
